### PR TITLE
chore: clean up project warnings

### PR DESCRIPTION
**Pull request by miketheman**
TL,DR: resolve outstanding warnings from mkdocs build and update runtime

See individual commits for specifics.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
 
 RUN set -x \
     && python3 -m venv /opt/venv
 
 ENV PATH="/opt/venv/bin:${PATH}"
 
-RUN pip --no-cache-dir --disable-pip-version-check install \
-    --upgrade pip setuptools wheel
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1
+ENV PIP_NO_CACHE_DIR=1
```
Comment (hugovk) [inline Dockerfile:17]: Not sure if this makes much difference?	
```suggestion
ARG PIP_DISABLE_PIP_VERSION_CHECK=1
ARG PIP_NO_CACHE_DIR=1
```

* `ARG` - only available during the build
* `ENV` - also available inside the image

https://vsupalov.com/docker-arg-env-variable-guide/#arg-and-env
https://docs.docker.com/build/building/variables/

**Review by hugovk — APPROVED** [review]
Thanks!

**Comment by ewdurbin** [comment]
What linter is being run? Can that be part of CI somehow?

**Comment by miketheman** [comment]
> What linter is being run? Can that be part of CI somehow?

No linter, rather mkdocs build or serve emits warnings that have been corrected.

**Comment by hugovk** [comment]
Could mirror the commands from https://github.com/psf/policies/blob/main/.readthedocs.yml in GitHub Actions with `--strict` to turn warnings into errors. 

https://www.mkdocs.org/user-guide/cli/#mkdocs-build

**Pull request merged**